Sources/StreamChatUI/Utils/UIImageView+Extensions.swift

```diff
@@ -6,10 +6,17 @@ import Nuke
 import UIKit
 
 extension UIImageView {
-    func loadImage(from url: URL?, placeholder: UIImage? = nil) {
+    func loadImage(from url: URL?, placeholder: UIImage? = nil, resizeAutomatically: Bool = true) {
         // Cancel any previous loading task
         currentImageLoadingTask?.cancel()
+
         guard let url = url else { image = nil; return }
+
+        let preprocessors: [ImageProcessing] = resizeAutomatically && bounds.size != .zero
+            ? [ImageProcessors.Resize(size: bounds.size, contentMode: .aspectFill, crop: true)]
```

Comment:
🤔 Shouldn't this also be somehow tested? Because it's out of Nuke scope basically and we have some custom logic in here....  But maybe it's too much :D I ask because this is not implemented in any tests AFAIK, so there is no way to know if this changes something visually . But on the other hand this would be probably more complex (and probably unnecessary work?) to test (Different set of Images, already tested by `Nuke`...)

Reply:
Or maybe the `UITests` already discovered :trollface:

Reply:
Ideally yes, but I actually did the opposite in 17ece73 to bypass Nuke completely. We need to have the image set synchronously. Previously we pre-loaded the image to Nuke's cache to force it to load synchronusly. With the various sizes now this becomes more complex so I removed it. So yeah, ideally we will have this tested eventually but I'm not sure exactly how to approach this. Also, it's not that big of a deal as long as all image setting logic is funneled via this method.